featurebyte/routes/observation_table/controller.py

```diff
@@ -149,10 +153,52 @@ async def update_observation_table(
         data: ObservationTableUpdate
             ObservationTable update payload
 
+        Raises
+        ------
+        ObservationTableInvalidUseCaseError
+            if use_case_ids is provided in the payload or use_case_id_to_add/use_case_id_to_remove is invalid
+
         Returns
         -------
         Optional[ObservationTableModel]
         """
+
+        if data.use_case_ids:
+            raise ObservationTableInvalidUseCaseError("use_case_ids is not a valid field to update")
+
+        if data.use_case_id_to_add or data.use_case_id_to_remove:
+            observation_table = await self.observation_table_service.get_document(
+                document_id=observation_table_id
+            )
+            use_case_ids = observation_table.use_case_ids if observation_table.use_case_ids else []
```

Comment:
If we change `use_case_ids` to `List[PydanticObjectId]`, we can simplify this part to `use_case_ids = observation_table.user_ids`.

Reply:
I think it make senses. An observation table cannot be attached to a UseCase unless it has been associated with a Context already